### PR TITLE
:bug: fix index out of range panic when annotation is not in 'key=value' format

### PR DESCRIPTION
**Pull request by lianghao208**
When using 
```
+kubebuilder:metadata:annotations="api-approved.kubernetes.io=xxx"
``` 
#691 in CRD struct files to generate api-approved annotation, if the annotation is not in `metadata:annotations:"xxx=xxx"` format, eg: 
```
+kubebuilder:metadata:annotations="api-approved.kubernetes.io: xxx"
```

 it triggers index out of range panic.

### CODE REVIEW

**Comment by JoelSpeed** [comment]
Code changes look ok, are there any test cases for the annotation marker that we can update to make sure this is working as expected/we don't regress in the future?

**Review by camilamacedo86 — APPROVED** [review]
That shows great 🥇 

/lgtm

**Comment by lianghao208** [comment]
> Code changes look ok, are there any test cases for the annotation marker that we can update to make sure this is working as expected/we don't regress in the future?
@JoelSpeed I will add some test cases to cover this scenario

**Comment by lianghao208** [comment]
@JoelSpeed I've add some test cases to cover, PTAL, thanks!

File: pkg/crd/parser_integration_test.go [hunk 1]
```diff
@@ -52,6 +53,18 @@ func packageErrors(pkg *loader.Package, filterKinds ...packages.ErrorKind) error
 	return outErr
 }
 
+func packageErrorsWithMessage(pkg *loader.Package, errMsg string) error {
+	var outErr error
```
Comment (JoelSpeed) [inline pkg/crd/parser_integration_test.go:57]: Do you need a slice of errors to collect all errors? Looks like for now you'll only collect the last package error?
Are we expecting multiple packages to have the same error?

File: pkg/crd/parser_integration_test.go [hunk 2]
```diff
@@ -132,6 +145,18 @@ var _ = Describe("CRD Generation From Parsing to CustomResourceDefinition", func
...
+		assertError := func(pkg *loader.Package, kind, errorMsg string) {
+			By(fmt.Sprintf("requesting that the %s CRD be generated", kind))
+			groupKind := schema.GroupKind{Kind: kind, Group: "testdata.kubebuilder.io"}
+			parser.NeedCRDFor(groupKind, nil)
+
+			By(fmt.Sprintf("fixing top level ObjectMeta on the %s CRD", kind))
+			crd.FixTopLevelMetadata(parser.CustomResourceDefinitions[groupKind])
+
+			By("checking that specific errors occurred along the way")
+			Expect(packageErrorsWithMessage(pkg, errorMsg)).To(HaveOccurred())
```
Comment (JoelSpeed) [inline pkg/crd/parser_integration_test.go:157]: An alternative suggestion, you could extract all errors and do something like, minor, depends if you prefer the readability of gomega or not
```
Expect(packageErrors(pkg)).To(ContainElement(MatchError(ContainSubstring(errorMsg))))
```

File: pkg/crd/parser_integration_test.go (hunk 1 above)
Comment (lianghao208) [reply pkg/crd/parser_integration_test.go:57]: In this case, it only needs to collect the last package error to assert just like `func packageErrors()`. We haven't met the case to expect multiple packages having different errors yet.

File: pkg/crd/parser_integration_test.go (hunk 2 above)
Comment (lianghao208) [reply pkg/crd/parser_integration_test.go]: Good suggestion and done :)

**Review by camilamacedo86 — APPROVED** [review]
@JoelSpeed,

Great point 👍 I think all is good to fly now.

/lgtm

**Review by JoelSpeed — COMMENTED** [review]
/lgtm

**Pull request merged**